### PR TITLE
Prevent the removal of a quest owner from a group, take 2

### LINKED ISSUE #12398: party leader should not be able to remove the owner of the party's quest
_I'm marking this as suggestion-discussion for a few days for the staff to comment on whether my suggested fix is desired and then if there's no objections I'll mark it as help wanted._

Currently, it is possible for a Party leader to remove the Party member who is the owner of the Party's current Quest. This leaves the Party's `quest` object in an invalid state (see the sample data at the bottom of this post). As a side-effect of this, the former Quest owner is not able to use any third-party tools that read or display party data (because the data is invalid) and they probably see errors in Habitica's official clients too.

Another side effect is that the former Quest owner is no longer able to control the quest (e.g., they can't abort it) and that means they have lost the Quest scroll and cannot gain the Quest rewards. Since the Quest scroll may have cost Gems, this is a significant problem.

I believe the best fix is that the API's [/api/v3/groups/:groupId/removeMember/:memberId](https://habitica.com/apidoc/#api-Group-RemoveGroupMember) route must throw an error if the Party leader tries to remove the Quest owner. This should happen regardless of whether the Quest is in the invitation stage or is active.

The error message could say something like "You cannot remove this member because they are the owner of your Party's current Quest. If you abort the Quest first, you can then remove this member." 

I'll leave this as suggestion-discussion for a few days for the staff to comment on whether this is the desired fix and then if there's no objections I'll mark it as help wanted.

NB It would be possible for the `removeMember` route to be changed so that it automatically aborted the Quest before removing the former Quest owner, but I believe that would be less desirable. The Party leader may actually not want to remove the member once they realise the member is the Quest owner. 

---

As an example of the invalid data, below is a `quest` object after the Quest owner and sole Quest participant was removed from the Party. The Quest is active but there is no `quest.leader` or `quest.key` field.

```
    "quest": {
        "progress": {
            "collect": {},
            "hp": 100
        },
        "active": true,
        "members": {
            "0d0fd796-3271-43f0-b042-10bfd7ed3d97": true
        },
        "extra": {}
    },
```

### PR DESCRIPTION
**Pull request by thewilloftheshadow**
Remake of #12680 but with the Pull app uninstalled

[//]: # (Note: See http://habitica.fandom.com/wiki/Using_Your_Local_Install_to_Modify_Habitica%27s_Website_and_API for more info)

[//]: # (Put Issue # here, if applicable. This will automatically close the issue if your PR is merged in)
Fixes #12398

### Changes
[//]: # (Describe the changes that were made in detail here. Include pictures if necessary)
This change will throw an error for the `/groups/:groupId/removeMember/:memberId` API route if the user that is being removed is the owner of the quest
This change adds a new string for the new error with the key `cannotRemoveQuestOwner`

[//]: # (Put User ID in here - found on the Habitica website at User Icon > Settings > API)

----
UUID: 2eb203f5-1b1e-4e5d-84a4-22f130a1090e

### CODE REVIEW

**Review by paglias — CHANGES_REQUESTED** [review]
@thewilloftheshadow thanks for the PR! The code is looking good, could you add an integration test that covers this new behavior? See https://github.com/HabitRPG/habitica/blob/develop/test/api/v3/integration/groups/POST-groups_id_removeMember.test.js and let me know if you need any help

**Comment by thewilloftheshadow** [comment]
It's been a while since I've been written any tests but I'll give it a go!

**Comment by thewilloftheshadow** [comment]
Ok, I wrote a test, and it ran both on my machine and [on the CI tests](https://github.com/HabitRPG/habitica/pull/12695/checks?check_run_id=1286198229#step:8:621)

File: website/server/controllers/api-v3/groups.js [hunk 1]
```diff
@@ -911,6 +911,8 @@ function _sendMessageToRemoved (group, removedUser, message, isInGroup) {
  * @apiError (400) {NotAuthorized} onlyLeaderCanRemoveMember Only the group
                                                              leader can remove members.
  * @apiError (400) {NotAuthorized} memberCannotRemoveYourself Group leader cannot remove themselves
+ * @apiError (400) {NotAuthorized} cannotRemoveQuestOwner Group leader cannot remove
```
Comment (paglias) [inline website/server/controllers/api-v3/groups.js]: the code here is 401 not 400, I think there was a mistake in the previous docs. NotAuthorized is 401, BadRequest is 400

File: test/api/v3/integration/groups/POST-groups_id_removeMember.test.js [hunk 2]
```diff
@@ -251,6 +251,29 @@ describe('POST /groups/:groupId/removeMember/:memberId', () => {
...
+      await party.sync();
+
+      expect(party.quest.members[partyLeader._id]).to.be.true;
+      expect(party.quest.members[partyMember._id]).to.be.true;
+
+      await party.sync();
+
+      expect(leader.post(`/groups/${party._id}/removeMember/${partyMember._id}`))
+        .to.eventually.be.rejected.and.eql({
+          code: 403,
```
Comment (paglias) [inline test/api/v3/integration/groups/POST-groups_id_removeMember.test.js]: code should be 401,

```
      code: 401,
      error: 'NotAuthorized',
      message: t('cannotRemoveQuestOwner'),
```

**Review by paglias — CHANGES_REQUESTED** [review]
two small issues then it's ready to go!

**Comment by thewilloftheshadow** [comment]
Fixed, and I changed the other NotAuthorized codes to 401 in the apidoc headers as well

**Comment by paglias** [comment]
@thewilloftheshadow  thanks for this PR! I'm merging the PR and awarding your contributor tier.

Note that further tiers require increasing amounts of work from one to the next, so Tier 2 may require a couple of PRs or a larger PR to attain. But keep helping out and we'll express our gratitude accordingly!

**Comment by thewilloftheshadow** [comment]
Awesome, thanks! I definitely plan on sticking around, that was fun!

**Pull request merged**